Silicon/CoffeelakePkg/Library/FirmwareUpdateLib/FirmwareUpdateLib.c

```diff
@@ -142,9 +147,11 @@ InitCsmeUpdInputData (
     CsmeUpdDriverInput->CompareMem       = (VOID *)((UINTN)CompareMem);
     CsmeUpdDriverInput->Stall            = (VOID *)((UINTN)MicroSecondDelay);
     CsmeUpdDriverInput->PciRead          = (VOID *)((UINTN)PciReadBuffer);
-    CsmeUpdDriverInput->HeciReadMessage  = (VOID *)((UINTN)HeciReceive);
-    CsmeUpdDriverInput->HeciSendMessage  = (VOID *)((UINTN)HeciSend);
-    CsmeUpdDriverInput->HeciReset        = (VOID *)((UINTN)HeciResetInterface);
+    if (HeciService != NULL) {
+      CsmeUpdDriverInput->HeciReadMessage  = (VOID *)((UINTN)HeciService->HeciReceive);
+      CsmeUpdDriverInput->HeciSendMessage  = (VOID *)((UINTN)HeciService->HeciSend);

```

Comment:
Since FirmwareUpdate doesn't link with HeciLib, CsmeUpdDriverInput->HeciReadMessage might be NULL if the service is not available. do you have that check before calling HeciReadMessage ()?

Reply:
These functions would be called from Csme Update library, we have no control over that code. 

I updated new patch to check if these are NULL and return NULL. In the calling function, it would return Unsupported when this function returns NULL

Reply:
thanks for the update.